src/utils/rollout_utils.ts

```diff
@@ -0,0 +1,30 @@
+import { createHash } from 'crypto';
+
+/**
+ * A function that allows for gradual, consistent rollouts of features
+ *
+ * Given a message, and a target rollout threshold  (between 0 and 1), returns a boolean of whether
+ * the message should get the rolled out treatment or not.
+ *
+ * This method has the property that the same message will always recieve the same treatment, even as
+ * the threshold is increased over time.
+ */
+export const isRolledOut = ({ message, threshold }: { message: string; threshold: number }): boolean => {
```

Comment:
You can make `message` a more generic type and allow this to work with a broader range of inputs:

https://www.typescriptlang.org/play?ssl=7&ssc=1&pln=7&pc=19#code/GYVwdgxgLglg9mABAZygJxmA5gSSgCgA8AuRAb0SjgGV1MtT8BKRAXgD4U7tEBfJ0qgw8yAKEQTEaAKZQQaJIQB0VWsKzMA3KN6jRQ+nnwAiABbSANhbjGm2g9iMBGAAx393XAQDaTgDQATH4AzAC6dkA